src/Middleware/DetectLocaleMiddleware.php

```diff
@@ -171,14 +188,17 @@ protected function setPersistLocale(HTTPRequest $request, $locale)
         }
 
         // Save locale
-        if ($locale) {
-            $request->getSession()->set($key, $locale);
-        } else {
-            $request->getSession()->clear($key);
+        $session = $request->getSession();
+        if ($session->isStarted()) {
```

Comment:
Missing ->getAll() check; A started session isn't valid if it's empty. I can add this in. It was in my pasted code.

Reply:
All sessions are started on every request, by the way.

Reply:
PR at https://github.com/tractorcow-farm/silverstripe-fluent/pull/476